### ISSUE TITLE
App list is empty again

### ISSUE THREAD

**Issue by SainoNamkho** [MuntashirAkon/AppManager#1297]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

Same as #1196, once been fixed, but comes back after same system or app update, however downgrading to the previous version does not help.

### To Reproduce

1. Reinstall the app
2. Find the list empty

### Screenshots

![Screenshot_2024-02-26-02-39-42-261_io github muntashirakon AppManager debug](https://github.com/MuntashirAkon/AppManager/assets/23036788/0b0457db-e59c-47b5-b494-5d7eb5762470)

### Logs

NonRoot log
```
2024-02-26 02:29:14.241 D/SplashActivity: Waiting to be authenticated.
2024-02-26 02:29:14.248 W/KSManager: Using keystore BKS
2024-02-26 02:29:14.250 I/CompatUtil: Loading local protection key
2024-02-26 02:29:14.253 I/CompatUtil: AES local protection key found in keystore
2024-02-26 02:29:14.266 D/SplashActivity: Authenticated
2024-02-26 02:29:14.268 D/SecurityAndOpsViewModel: Before Ops::init
2024-02-26 02:29:14.269 D/SecurityAndOpsViewModel: After Ops::init
2024-02-26 02:29:14.346 D/SplashActivity: Authentication completed.
2024-02-26 02:29:14.390 D/BaseActivity: Already authenticated.
2024-02-26 02:29:14.500 D/MVM: New instance created
2024-02-26 02:29:14.639 D/PackageUtils: Loading apps for the first time.
2024-02-26 02:29:14.723 D/Users: Missing required permission: MANAGE_USERS or CREATE_USERS (7+). Falling back to unprivileged mode.
2024-02-26 02:29:15.101 W/AppDb: Could not retrieve package info list for user 0
android.os.BadParcelableException: Failure retrieving array; only received 114 of 426
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
	at android.os.Parcel.readTypedObject(Parcel.java:4025)
	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda8.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(BinderProxy.java:639)
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
	... 18 more
2024-02-26 02:29:15.367 W/AppDb: Could not retrieve package info list for user 999
android.os.BadParcelableException: Failure retrieving array; only received 102 of 426
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
	at android.os.Parcel.readTypedObject(Parcel.java:4025)
	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda8.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(BinderProxy.java:639)
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
	... 18 more
2024-02-26 02:29:15.604 W/AppUsageStatsManager: Missing required permission: android.permission.READ_PRIVILEGED_PHONE_STATE
2024-02-26 02:29:15.861 W/ExUtils: (Suppressed error)
java.lang.SecurityException: No permission to query usage stats for this user: uid 10329 does not have android.permission.INTERACT_ACROSS_USERS_FULL.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
	at android.os.Parcel.createException(Parcel.java:3041)
	at android.os.Parcel.readException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:2966)
	at android.app.usage.IUsageStatsManager$Stub$Proxy.queryEventsForUser(IUsageStatsManager.java:972)
	at io.github.muntashirakon.AppManager.compat.UsageStatsManagerCompat.queryEvents(UsageStatsManagerCompat.java:36)
	at io.github.muntashirakon.AppManager.usage.AppUsageStatsManager.getUsageStatsInternal(AppUsageStatsManager.java:237)
	at io.github.muntashirakon.AppManager.usage.AppUsageStatsManager.getUsageStats(AppUsageStatsManager.java:169)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.lambda$updateVariableData$0(AppDb.java:367)
	at io.github.muntashirakon.AppManager.db.utils.AppDb$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at io.github.muntashirakon.AppManager.utils.ExUtils.exceptionAsNull(ExUtils.java:77)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateVariableData(AppDb.java:366)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:295)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda8.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2343)
	at android.app.ContextImpl.enforceCallingPermission(ContextImpl.java:2361)
	at com.android.server.usage.UsageStatsService$BinderService.queryEventsForUser(UsageStatsService.java:2266)
	at android.app.usage.IUsageStatsManager$Stub.onTransact(IUsageStatsManager.java:447)
	at android.os.Binder.execTransactInternal(Binder.java:1346)

2024-02-26 02:29:17.347 D/BaseActivity: Already authenticated.
```
***
Root log
```
2024-02-26 02:42:50.177 D/SplashActivity: Waiting to be authenticated.
2024-02-26 02:42:50.179 W/KSManager: Using keystore BKS
2024-02-26 02:42:50.180 I/CompatUtil: Loading local protection key
2024-02-26 02:42:50.187 I/CompatUtil: AES local protection key found in keystore
2024-02-26 02:42:50.200 D/SplashActivity: Authenticated
2024-02-26 02:42:50.202 D/SecurityAndOpsViewModel: Before Ops::init
2024-02-26 02:42:50.206 D/IPC: Init: Local server
2024-02-26 02:42:50.283 D/ServiceConnectionWrapper: Launching service...
2024-02-26 02:42:50.320 D/Runner: RootShell
2024-02-26 02:42:50.569 D/ServiceConnectionWrapper: service onServiceConnected: ComponentInfo{io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.ipc.AMService}
2024-02-26 02:42:50.570 D/ServiceConnectionWrapper: Launching service...
2024-02-26 02:42:50.572 D/ServiceConnectionWrapper: service onServiceConnected: ComponentInfo{io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.ipc.FileSystemService}
2024-02-26 02:42:50.578 D/SecurityAndOpsViewModel: After Ops::init
2024-02-26 02:42:50.578 D/SplashActivity: Authentication completed.
2024-02-26 02:42:50.614 D/BaseActivity: Already authenticated.
2024-02-26 02:42:50.707 D/MVM: New instance created
2024-02-26 02:42:50.790 D/PackageUtils: Loading apps for the first time.
2024-02-26 02:42:51.239 W/AppDb: Could not retrieve package info list for user 0
android.os.BadParcelableException: Failure retrieving array; only received 114 of 427
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
	at android.os.Parcel.readTypedObject(Parcel.java:4025)
	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda8.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(BinderProxy.java:639)
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
	... 18 more
2024-02-26 02:42:51.496 W/AppDb: Could not retrieve package info list for user 999
android.os.BadParcelableException: Failure retrieving array; only received 102 of 427
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
	at android.os.Parcel.readTypedObject(Parcel.java:4025)
	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda8.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(BinderProxy.java:639)
	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
	... 18 more
```

### Device info

- Device: Xiaomi Redmi K70
- OS Version: HyperOS 1.0.14.0.UNKCNXM, Android14,  rooted by apatch
- App Manager Version: 3.1.4 and 4.0.0-alpha02-DEBUG (the latest from https://github.com/MuntashirAkon/AppManager/actions/runs/7970127720)
- Mode: Root/ADB/NonRoot

**Comment by SainoNamkho** [comment]
Seems some selinux context is broken on my phone. I'll try to reset my phone to fix it and see if the problem holds.

**Comment by MuntashirAkon** [comment]
Is there an update?

**Comment by cntvc** [comment]
I see the same issue on mobel phone.

- Device: Xiaomi 13
- OS: Xiaomi HyperOS, 1.0.8.0.UMCCNXM, Android14
- App Manager Version: 4.0.0-alpha02-DEBUG, 3.1.5 Stable
- Mode: NonRoot

log file: [file_803639665702338770.zip](https://github.com/MuntashirAkon/AppManager/files/14879102/file_803639665702338770.zip)

Besides that, I also sent the logs captured using the "adb logcat" command to @MuntashirAkon's email, hoping this would be helpful.

**Comment by SainoNamkho** [comment]
> Is there an update?

After a factory reset it works well now. Sorry for not having further clues to reproducing the problem.

**Comment by tiann** [comment]
Same issue here, i'll look into it:

```txt
2024-04-20 16:00:45.838 23701-23852 BpBinder                        W  Large or Failed outgoing transaction of 4 bytes, interface descriptor , code 1
2024-04-20 16:00:45.838 23701-23852 JavaBinder                      E  !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
2024-04-20 16:00:45.838 23701-23852 AppDb                           W  Could not retrieve package info list for user 0
                                                                       android.os.BadParcelableException: Failure retrieving array; only received 291 of 585
                                                                       	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
                                                                       	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
                                                                       	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
                                                                       	at android.os.Parcel.readTypedObject(Parcel.java:4025)
                                                                       	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
                                                                       	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0)
                                                                       	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
                                                                       	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
                                                                       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                       	at java.lang.Thread.run(Thread.java:1012)
                                                                       Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
                                                                       	at android.os.BinderProxy.transactNative(Native Method)
                                                                       	at android.os.BinderProxy.transact(BinderProxy.java:621)
                                                                       	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
                                                                       	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42) 
                                                                       	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0) 
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80) 
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78) 
                                                                       	at android.os.Parcel.readTypedObject(Parcel.java:4025) 
                                                                       	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070) 
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166) 
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126) 
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260) 
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151) 
                                                                       	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151) 
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343) 
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0) 
                                                                       	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487) 
                                                                       	at java.util.concurrent.FutureTask.run(FutureTask.java:264) 
                                                                       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 
                                                                       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644) 
                                                                       	at java.lang.Thread.run(Thread.java:1012) 
2024-04-20 16:00:45.838 23701-23852 JavaBinder                      W  ibinderForJavaObject: 0x71ab27d140 is not a Binder object
2024-04-20 16:00:45.838 23701-23852 Parcel                          E  Native binder in markForBinder is null for non-null jobject
2024-04-20 16:00:45.839  8483-23916 GameBoosterService              I  restartGameMode: io.github.muntashirakon.AppManager
2024-04-20 16:00:45.846  2158-9599  CoreBackPreview                 D  Window{bd7cea7 u0 Splash Screen io.github.muntashirakon.AppManager EXITING}: Setting back callback null
2024-04-20 16:00:45.847  2158-7251  InputManager-JNI                W  Input channel object 'bd7cea7 Splash Screen io.github.muntashirakon.AppManager (client)' was disposed without first being removed with the input manager!
2024-04-20 16:00:45.852 23701-23852 JavaBinder                      W  ibinderForJavaObject: 0x71ab27d140 is not a Binder object
2024-04-20 16:00:45.852 23701-23852 Parcel                          E  Native binder in markForBinder is null for non-null jobject
2024-04-20 16:00:45.870 23774-23849 io.github.munt...anager:priv:0  W  Large reply transaction of 340380 bytes, interface descriptor io.github.muntashirakon.AppManager.IAMService, code 2
2024-04-20 16:00:45.920 23701-23852 IPCThreadState                  E  Binder transaction failure: 89290836/29201/-28
2024-04-20 16:00:45.920 23701-23852 BpBinder                        W  Large or Failed outgoing transaction of 4 bytes, interface descriptor , code 1
2024-04-20 16:00:45.920 23701-23852 JavaBinder                      E  !!! FAILED BINDER TRANSACTION !!!  (parcel size = 4)
2024-04-20 16:00:45.921 23701-23852 AppDb                           W  Could not retrieve package info list for user 999
                                                                       android.os.BadParcelableException: Failure retrieving array; only received 291 of 585
                                                                       	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:104)
                                                                       	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42)
                                                                       	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0)
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80)
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78)
                                                                       	at android.os.Parcel.readTypedObject(Parcel.java:4025)
                                                                       	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070)
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166)
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
                                                                       	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0)
                                                                       	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
                                                                       	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
                                                                       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                       	at java.lang.Thread.run(Thread.java:1012)
                                                                       Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died, but this could also be caused by running out of binder buffer space
                                                                       	at android.os.BinderProxy.transactNative(Native Method)
                                                                       	at android.os.BinderProxy.transact(BinderProxy.java:621)
                                                                       	at android.content.pm.BaseParceledListSlice.<init>(BaseParceledListSlice.java:95)
                                                                       	at android.content.pm.ParceledListSlice.<init>(ParceledListSlice.java:42) 
                                                                       	at android.content.pm.ParceledListSlice.<init>(Unknown Source:0) 
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:80) 
                                                                       	at android.content.pm.ParceledListSlice$1.createFromParcel(ParceledListSlice.java:78) 
                                                                       	at android.os.Parcel.readTypedObject(Parcel.java:4025) 
                                                                       	at android.content.pm.IPackageManager$Stub$Proxy.getInstalledPackages(IPackageManager.java:5070) 
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackagesInternal(PackageManagerCompat.java:166) 
                                                                       	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:126) 
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:260) 
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151) 
                                                                       	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151) 
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343) 
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0) 
                                                                       	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487) 
                                                                       	at java.util.concurrent.FutureTask.run(FutureTask.java:264) 
                                                                       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 
                                                                       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644) 
                                                                       	at java.lang.Thread.run(Thread.java:1012) 
2024-04-20 16:00:45.924 23701-23852 UriManager                      W  Failed reading Uri grants
                                                                       java.io.IOException: Exception thrown on remote process
                                                                       	at io.github.muntashirakon.io.IOResult.checkException(IOResult.java:34)
                                                                       	at io.github.muntashirakon.io.RemoteFile.newInputStream(RemoteFile.java:429)
                                                                       	at io.github.muntashirakon.io.AtomicExtendedFile.openRead(AtomicExtendedFile.java:173)
                                                                       	at io.github.muntashirakon.AppManager.uri.UriManager.readGrantedUriPermissions(UriManager.java:129)
                                                                       	at io.github.muntashirakon.AppManager.uri.UriManager.<init>(UriManager.java:63)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateVariableData(AppDb.java:358)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:295)
                                                                       	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)
                                                                       	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:151)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:343)
                                                                       	at io.github.muntashirakon.AppManager.main.MainViewModel$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0)
                                                                       	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
                                                                       	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
                                                                       	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                       	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                       	at java.lang.Thread.run(Thread.java:1012)
                                                                       Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
                                                                       	at libcore.io.Linux.open(Native Method)
                                                                       	at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
                                                                       	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:274)
                                                                       	at android.system.Os.open(Os.java:494)
                                                                       	at io.github.muntashirakon.io.FileSystemService.openReadStream(FileSystemService.java:303)
                                                                       	at io.github.muntashirakon.io.IFileSystemService$Stub.onTransact(IFileSystemService.java:597)
                                                                       	at android.os.Binder.execTransactInternal(Binder.java:1346)
                                                                       	at android.os.Binder.execTransact(Binder.java:1282)
                                                                       	
```

**Comment by MuntashirAkon** [comment]
Fixed via c47ce6fdeb05edce6de6dc76173de5bb08665d81
